### PR TITLE
Console history feature

### PR DESCRIPTION
**Pull request by afshin**
This PR is just meant to catch master up to the current state of the codebase. It adds a new console history class and history navigation signals.

cf. https://github.com/jupyter/jupyter-js-notebook/issues/170

### CODE REVIEW

File: src/console/history.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      return;
+    }
+    this._kernel = newValue;
+  }
+
+  /**
+   * Construct a new console history object.
+   */
+  constructor(kernel: IKernel) {
+    // Listen to the command history events of the model.
```
Comment (blink1073) [inline src/console/history.ts]: Is this a TODO?

Comment (afshin) [inline src/console/history.ts]: Sorry, I wasn't clear in the comments up top. This is just to open the PR for this feature and to keep other watchers apprised.

**Comment by afshin** [comment]
history

**Comment by afshin** [comment]
One known bug is the inconsistency of setting the `cursorPosition` of the prompt to the the end of the text when there is whitespace in the text field. Perhaps we need a flag value for `last` that we can set that model to or some other way. This isn't necessarily a history bug so I'm requesting review and merge on this PR.

File: src/console/history.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+export
+class ConsoleHistory implements IConsoleHistory {
+  /**
+   * Get whether the console history manager is disposed.
+   *
+   * #### Notes
+   * This is a read-only property.
+   */
+  get isDisposed(): boolean {
+    return false;
```
Comment (sccolbert) [inline src/console/history.ts]: never disposed?

Comment (afshin) [inline src/console/history.ts]: Scaffolding bug! Thanks.

**Comment by blink1073** [comment]
Noice!

**Pull request merged**